tests/Library/Garden/BasicSchemaTest.php

```diff
@@ -346,6 +346,57 @@ public function provideBooleanData() {
     }
 
     /**
+     * Call validate on an instance of Schema where the data contains unexpected parameters.
+     * *
+     * @param int $validationBehavior
+     */
+    protected function doValidationBehavior($validationBehavior) {
+        $schema = new Schema([
+            'i:userID' => 'The ID of the user.',
+            's:name' => 'The username of the user.',
+            's:email' => 'The email of the user.',
+        ]);
+        $schema->setValidationBehavior($validationBehavior);
+
+        $data = [
+            'userID' => 123,
+            'name' => 'foo',
+            'email' => 'user@example.com',
+            'admin' => true,
+            'role' => 'Administrator'
+        ];
+
+        $schema->validate($data);
+        $this->assertArrayNotHasKey('admin', $data);
+        $this->assertArrayNotHasKey('role', $data);
+    }
+
+    /**
+     * Test throwing an exception when removing unexpected parameters from validated data.
+     *
+     * @expectedException \Garden\Exception\ValidationException
+     */
+    public function testValidateException() {
```

Comment:
You could do all these three tests with an @dataProvider attribute on the original test.